apps/heft/src/cli/actions/BuildAction.ts

```diff
@@ -6,15 +6,96 @@ import {
   CommandLineStringParameter,
   ICommandLineActionOptions
 } from '@rushstack/ts-command-line';
+import { SyncHook, AsyncParallelHook, AsyncSeriesHook } from 'tapable';
+import { performance } from 'perf_hooks';
 
 import { HeftActionBase, IHeftActionBaseOptions, ActionHooksBase, IActionDataBase } from './HeftActionBase';
+import { CleanAction } from './CleanAction';
 
-export interface IBuildActionOptions extends IHeftActionBaseOptions {}
+export interface IBuildActionOptions extends IHeftActionBaseOptions {
+  cleanAction: CleanAction;
+}
+
+/**
+ * @public
+ */
+export class BuildPhaseHooksBase {
+  public readonly run: AsyncParallelHook = new AsyncParallelHook();
+}
+
+/**
+ * @public
+ */
+export interface IBuildPhase<TBuildPhaseHooks extends BuildPhaseHooksBase = BuildPhaseHooksBase> {
+  hooks: TBuildPhaseHooks;
+}
+
+/**
+ * @public
+ */
+export interface ISharedCopyStaticAssetsConfiguration {
+  /**
+   * File extensions that should be copied from the src folder to the destination folder(s)
+   */
+  fileExtensions?: string[];
```

Comment:
Can `fileExtensions` and `include` be specified together?  If so, how do they interact?

Reply:
They can. If they are, both are used. The idea is that you use `fileExtensions` to copy your images and you use `include` to either copy individual files or to copy sets of files (i.e. - in a generator project, you might say something like `templates/**/*`)

Reply:
I see, so `fileExtensions` is applied as a UNION with `include`, and then `exclude` applies to the union?

We should clarify that in the docs.